src/services/artifacts.ts

```diff
@@ -39,3 +39,30 @@ export const downloadArtifact = async (
   document.body.removeChild(a);
   window.URL.revokeObjectURL(blobUrl);
 };
+
+export const downloadArtifactArchive = async (fileName: string, executionId: string) => {
+  const url = `/executions/${executionId}/artifact-archive`;
+  const finalUrl = `${getApiEndpoint()}${getRtkBaseUrl(undefined)}${url}`;
+  const idToken = await getRtkIdToken();
+
+  // Call the API to retrieve file or signed URL
+  let response = await fetch(finalUrl, {
+    headers: idToken ? {authorization: `Bearer ${idToken}`} : {},
+  });
+
+  // When the signed URL is returned, follow it
+  if (response.headers.get('content-type') === 'application/json') {
+    response = await fetch((await response.json()).data.url);
+  }
+
+  // Download the file
+  const blobUrl = window.URL.createObjectURL(await response.blob());
+  const a = document.createElement('a');
+  a.href = blobUrl;
+  // tar.gz format for archive
+  a.download = `${fileName}.tar.gz`;
```

Comment:
Shouldn't the `fileName` have the extension itself?

Reply:
I thought I shouldn't pollute Artifacts list component with it. For example if it would be required to download archive from some other place, we would have to not forget about specifying this format.

Reply:
I don't think that it's pollution, the extension is part of the file name. The archive in different places could be of a different format (in theory, because in practice I doubt so)

Reply:
Ok